### PR TITLE
Volunteer input validation

### PR DESCRIPTION
**Pull request by csiyang**
- refactored form with Formik
- added validation to user input
- updated tests
- add new tests for validation

Note: I think there's something broken with the form on consume

### CODE REVIEW

File: client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js [hunk 1]
```diff
@@ -1,43 +1,71 @@
...
+    render(<AddVolunteerForm onSubmit={handleSubmit}/>)
+
+    userEvent.clear(screen.getByLabelText(/firstName/i))
+    userEvent.clear(screen.getByLabelText(/lastName/i))
+
+    userEvent.click(screen.getByRole('button', { name: /add/i }))
+
+    const element = await screen.findAllByText('Required')
+
+    expect(addVolunteer).not.toHaveBeenCalled()
```
Comment (don-smith) [inline client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js]: I don't think this is asserting anything important. The only reason it isn't being called is because you're mocking out the event handler. I would remove this line and line 7. I don't think there is any reason you need to import `addVolunteer`.

File: client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js [hunk 2]
```diff
@@ -1,43 +1,71 @@
...
+    expect(element[0]).toBeInTheDocument()
```
Comment (don-smith) [inline client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js]: Because you're expecting line 22 to return 2 elements, it would probably be a good idea to `expect(elements).toHaveLength(2)`.

File: client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js [hunk 3]
```diff
@@ -1,43 +1,71 @@
...
+  it('Should display "required" for empty required entires', async () => {
+    const handleSubmit = jest.fn()
+    render(<AddVolunteerForm onSubmit={handleSubmit}/>)
+
+    userEvent.clear(screen.getByLabelText(/firstName/i))
+    userEvent.clear(screen.getByLabelText(/lastName/i))
+
+    userEvent.click(screen.getByRole('button', { name: /add/i }))
+
+    const element = await screen.findAllByText('Required')
```
Comment (don-smith) [inline client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js]: I would rename `element` to `elements` because it will be an array.

File: server/.env.example [hunk 4]
```diff
@@ -1,5 +1,5 @@
 JWT_SECRET='dummy secret'
 SENDGRID_API_KEY='replace-with-api-key-from-sendgrid'
-AUTH0_DOMAIN = 'get the domain from auth0.con'
-AUTH0_API_EXPLORER_CLIENTID = 'get the id from auth0.con'
-AUTH0_API_EXPLORER_SECRET= 'get the id from auth0.com' 
\ No newline at end of file
+AUTH0_DOMAIN = 'https://gardenz.au.auth0.com/'
+AUTH0_API_EXPLORER_CLIENTID = '2cxnaycUS9BKgNlesVGlfIGIQZjVxPTu'
+AUTH0_API_EXPLORER_SECRET= 'WSWxcWFQNZWgUFn_ojRwYuraFkynxs6yFwu-SHMnl1AvzXt4JE-cfN2y-B5Aw_Pa'
```
Comment (don-smith) [inline server/.env.example]: These changes should only be made to your local `.env` file ... NOT `.env.example`. This file will be committed and it now contains secrets 😬 Just undo the changes to this file.

File: server/routes/auth.js [hunk 5]
```diff
@@ -7,7 +7,9 @@ const clientId = process.env.AUTH0_API_EXPLORER_CLIENTID
 const secret = process.env.AUTH0_API_EXPLORER_SECRET
 
 const getUserRoles = async (uid) => {
+  console.log('domain:', domain)
   const accessToken = await getAccessToken()
+  console.log(accessToken)
```
Comment (don-smith) [inline server/routes/auth.js]: Please remove these console.logs

**Review by don-smith — CHANGES_REQUESTED** [review]

File: server/.env.example (hunk 4 above)
Comment (csiyang) [reply server/.env.example]: Oops!

File: server/routes/auth.js (hunk 5 above)
Comment (csiyang) [reply server/routes/auth.js]: Updated

File: client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js (hunk 3 above)
Comment (csiyang) [reply client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js]: Updated

File: client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js (hunk 2 above)
Comment (csiyang) [reply client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js]: Updated

File: client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js (hunk 1 above)
Comment (csiyang) [reply client/components/volunteers/RockUpVolunteerForm/AddVolunteerForm.test.js]: Updated

**Comment by csiyang** [comment]
- Updated how error messages are displayed
- Made requested changes
- changed classNames for form fields

**Review by don-smith — APPROVED** [review]
